prometheus_client/multiprocess.py

```diff
@@ -37,142 +80,145 @@ def __init__(self, registry, path=None):
-                if metric.type == Gauge._type:
-                    without_pid = tuple(l for l in labels if l[0] != 'pid')
-                    if metric._multiprocess_mode == Gauge.MIN:
-                        current = samples.setdefault((name, without_pid), value)
-                        if value < current:
-                            samples[(s.name, without_pid)] = value
-                    elif metric._multiprocess_mode == Gauge.MAX:
-                        current = samples.setdefault((name, without_pid), value)
-                        if value > current:
-                            samples[(s.name, without_pid)] = value
-                    elif metric._multiprocess_mode == Gauge.LIVESUM:
-                        samples[(name, without_pid)] += value
-                    else:  # all/liveall
-                        samples[(name, labels)] = value
-
-                elif metric.type == 'histogram':
-                    bucket = tuple(float(l[1]) for l in labels if l[0] == 'le')
-                    if bucket:
-                        # _bucket
-                        without_le = tuple(l for l in labels if l[0] != 'le')
-                        buckets.setdefault(without_le, {})
-                        buckets[without_le].setdefault(bucket[0], 0.0)
-                        buckets[without_le][bucket[0]] += value
-                    else:
-                        # _sum/_count
-                        samples[(s.name, labels)] += value
-                else:
-                    # Counter and Summary.
-                    samples[(s.name, labels)] += value
-
-
-            # Accumulate bucket values.
-            if metric.type == 'histogram':
-                for labels, values in buckets.items():
-                    acc = 0.0
-                    for bucket, value in sorted(values.items()):
-                        sample_key = (
-                            metric.name + '_bucket',
-                            labels + (('le', floatToGoString(bucket)),),
-                        )
-                        if accumulate:
-                            acc += value
-                            samples[sample_key] = acc
-                        else:
-                            samples[sample_key] = value
-                    if accumulate:
-                        samples[(metric.name + '_count', labels)] = acc
-            # Convert to correct sample format.
-            metric.samples = [Sample(name_, dict(labels), value) for (name_, labels), value in samples.items()]
-        return metrics.values()
 
     def collect(self, blocking=True):
         # blocking is used for testing purposes
         lock_type = LOCK_SH if blocking else LOCK_SH | LOCK_NB
         with advisory_lock(lock_type):
             files = glob.glob(os.path.join(self._path, '*.db'))
-            return self.merge(files, accumulate=True)
+            return merge(files, accumulate=True)
+
+
+def merge(files, accumulate=True):
```

Comment:
It would be great if we can split the file processing part out of this function and make it simply about `merging` the various values in each